addons/l10n_it_edi_extended/models/account_edi_format.py

```diff
@@ -0,0 +1,91 @@
+        body_tree = vals.get('body_tree')
+        message_to_log = vals.get('message_to_log')
+
+        withholding_elements = body_tree.xpath('.//DatiGeneraliDocumento/DatiRitenuta')
+        withholding_taxes = []
+        for withholding in (withholding_elements or []):
+            tipo_ritenuta = withholding.find("TipoRitenuta")
+            reason = withholding.find("CausalePagamento")
+            percentage = withholding.find('AliquotaRitenuta')
+            withholding_type = tipo_ritenuta.text if tipo_ritenuta is not None else "RT02"
+            withholding_reason = reason.text if reason is not None else "A"
+            withholding_percentage = -float(percentage.text if percentage is not None else "0.0")
+            withholding_tax = self._l10n_it_edi_search_tax(company, withholding_percentage, l10n_it_withholding_type=withholding_type, l10n_it_withholding_reason=withholding_reason)
+            if withholding_tax:
+                withholding_taxes.append(withholding_tax)
+            else:
+                message = self._l10n_it_edi_missing_tax_message(percentage, l10n_it_withholding_type=withholding_type, l10n_it_withholding_reason=withholding_reason)
+                message_to_log.append(message)
+
+        pension_fund_elements = body_tree.xpath('.//DatiGeneraliDocumento/DatiCassaPrevidenziale')
+        pension_fund_taxes = []
+        for pension_fund in (pension_fund_elements or []):
+            pension_fund_type = pension_fund.find("TipoCassa")
+            tax_factor_percent = pension_fund.find("AlCassa")
+            vat_tax_factor_percent = pension_fund.find("AliquotaIVA")
+            pension_fund_type = pension_fund_type.text if pension_fund_type is not None else ""
+            tax_factor_percent = float(tax_factor_percent.text or "0.0")
+            vat_tax_factor_percent = float(vat_tax_factor_percent.text or "0.0")
+            pension_fund_tax = self._l10n_it_edi_search_tax(company, tax_factor_percent, l10n_it_pension_fund_type=pension_fund_type)
+            if pension_fund_tax:
+                pension_fund_taxes.append(pension_fund_tax)
+            else:
+                message = self._l10n_it_edi_missing_tax_message(tax_factor_percent, l10n_it_pension_fund_type=pension_fund_type)
+                message_to_log.append(message)
+        return {
+            **extra_info,
+            "withholding_taxes": withholding_taxes,
+            "pension_fund_taxes": pension_fund_taxes
+        }
+
+    def _import_fattura_pa_line(self, vals):
+        super()._import_fattura_pa_line(vals)
+        invoice_line_form = vals['invoice_line_form']
+        for withholding_tax in vals.get('withholding_taxes', []):
+            invoice_line_form.tax_ids.add(withholding_tax)
+        for pension_fund_tax in vals.get('pension_fund_taxes', []):
+            invoice_line_form.tax_ids.add(pension_fund_tax)
+
+    def _l10n_it_edi_missing_tax_message(self, percentage, name=False, **extra_domain):
```

Comment:
No need for separate method.  Also, we could even just say: check the configuration of this tax...